services-core/src/history/events.rs

```diff
@@ -93,9 +93,9 @@ impl EventRegistry {
         // Create temp file to be written completely before rename
         let temp_file = File::create(&temp_path)
             .with_context(|| format!("couldn't create {}", temp_path.display()))?;
-        let buffered_writer = BufWriter::new(temp_file);
-        self.write_to(buffered_writer)?;
-
+        let mut buffered_writer = BufWriter::new(temp_file);
+        self.write_to(&mut buffered_writer)?;
+        buffered_writer.flush()?;
```

Comment:
There might be an issue here with the fact that the `File` has not been `drop`-ed before the file is renamed? Maybe there should be an explicit scope so that the `buffered_writer` and its inner `File` get cleaned up:
```suggestion
        {
            let mut buffered_writer = BufWriter::new(temp_file);
            self.write_to(&mut buffered_writer)?;
            buffered_writer.flush()?;
        }
```